llvm/include/llvm/TargetParser/Triple.h

```diff
@@ -349,7 +350,12 @@ class Triple {
   /// triple fields unknown.
   Triple() = default;
 
+  LLVM_ABI explicit Triple(std::string &&Str);
+  explicit Triple(StringRef Str) : Triple(Str.str()) {}
+  explicit Triple(const char *Str) : Triple(std::string(Str)) {}
+  explicit Triple(const std::string &Str) : Triple(std::string(Str)) {}
```

Comment:
With the StringRef overload, is `Triple(const std::string &Str)` still useful?

Reply:
This avoids ambiguous conversion to Twine or StringRef